addons/friendly_tracker/functions/fnc_getTrackedSides.sqf

```diff
@@ -0,0 +1,33 @@
+#include "script_component.hpp"
+/*
+ * Author: 3Mydlo3
+ * Returns trackable sides for given side.
+ *
+ * Arguments:
+ * 0: Side for which we want trackable sides <SIDE>
+ *
+ * Return Value:
+ * 0: Array of trackable sides <ARRAY>
+ *
+ * Example:
+ * call afft_friendly_tracker_fnc_getTrackedSides
+ *
+ * Public: No
+ */
+
+params [["_side", playerSide]];
+
+private _trackedSides = switch (GVAR(showSides)) do {
+    case (SIDES_FRIENDLY): {([_side] call BIS_fnc_friendlySides) - [sideFriendly]};
+    case (SIDES_PLAYER): {[_side]};
+    case (SIDES_ALL): {[WEST, EAST, INDEPENDENT]};
+};
+
+if (GVAR(showSideCivilian)) then {
+    _trackedSides pushBackUnique CIVILIAN;
+} else {
+    if (_side isEqualTo CIVILIAN) exitWith {};
+    _trackedSides = _trackedSides - [CIVILIAN];
+};
+
+_trackedSides
```

Comment:
Original:
```
Result: 0.0238 ms
Cycles: 10000/10000
```
Changed:
```
Result: 0.0229 ms
Cycles: 10000/10000
```
```suggestion
params [["_side", playerSide]];

private _trackedSides = switch (GVAR(showSides)) do {
    case (SIDES_FRIENDLY): {([_side] call BIS_fnc_friendlySides) - [sideFriendly]};
    case (SIDES_PLAYER): {[_side, CIVILIAN]};
    case (SIDES_ALL): {[WEST, EAST, INDEPENDENT, CIVILIAN]};
};

if !(GVAR(showSideCivilian)) exitWith {
    _trackedSides - [CIVILIAN] // return
};

_trackedSides // return
```

Reply:
Could be made faster by doing custom implementation of `BIS_fnc_friendlySides` or inlining the functionality (macro?).

Reply:
I'll look into `BIS_fnc_friendlySides` and check if it can be implemented better. Not sure if it's even possible to make this function much faster as we need to cover these cases either way.

Reply:
It cannot be changed as you proposed. Civilians wouldn't see themselves if `GVAR(showSideCivilian) = false` and I don't think it's good behaviour. I thought about special civilians handling and maybe they need their own settings if should have tracker enabled and which sides should they see.

Reply:
Forgot to add one check: `(!GVAR(showSideCivilian) && {_side != CIVILIAN})` should fix the issue and still be faster on average.